src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,157 @@
+import React, {
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+
+import { TodoList } from '../TodoList';
+import { apiClient } from '../../api/todos';
+import { useTodosContext } from '../store';
+import { ErrorOption } from '../../enum/ErrorOption';
+import { TodoFooter } from '../TodoFooter';
+import { TodoItem } from '../TodoItem';
+import { Todo } from '../../types/Todo';
+
+const USER_ID = 12027;
+
+export const TodoApp: React.FC = () => {
+  const {
+    todos,
+    addTodo,
+    hasError,
+    setError,
+    recieveTodos,
+    resetHasError,
+    toggleAllTodoCondition,
+  } = useTodosContext();
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [inputValue, setInputValue] = useState('');
+  const [isPostRequest, setIsPostRequest] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, []);
+
+  useEffect(() => {
+    resetHasError();
+
+    apiClient.getTodos(USER_ID)
+      .then(todosFS => {
+        recieveTodos(todosFS);
+      })
+      .catch(() => setError(ErrorOption.RecivingError))
+      .finally(() => setIsLoading(false));
+  }, [resetHasError, recieveTodos, setError]);
+
+  const onSubmitForm = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (inputValue.trim() === '') {
```

Comment:
```suggestion
    if (inputValue.trim()) {
```

Reply:
That ok for reading ?

Maybe i have little experience, but for me `if (inputValue.trim() === '')` looks more readable 👀

Reply:
And also that is not works for me 👀 (probably i don't understand something)
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/67290782/385fafde-8b44-4985-99d6-5a414a16cf7d)

Reply:
> inputValue.trim()

it should be `!inputValue.trim()`